### DISCUSSION TITLE
Side-by-side images in Zim become  vertically aligned once exported to HTML

### DISCUSSION THREAD

**Discussion by aloboa** [zim-desktop-wiki/zim-desktop-wiki#2559] [Q&A]
I can have side-by-side images in Zim, but once exported to HTML I often get them vertically aligned.
Is there a way to arrange the images in HTML as in Zim?

![Zim_Screenshot](https://github.com/zim-desktop-wiki/zim-desktop-wiki/assets/6074054/df26ef57-d2ae-48b3-949a-ca8d758cd30f)

![HTML_Screenshot](https://github.com/zim-desktop-wiki/zim-desktop-wiki/assets/6074054/85c745aa-9b48-488f-b60f-11a116712db9)

**Comment by introt** [comment]
It looks like they're full size and simply don't fit on the same line. If that's the case (as opposed to a line break being inserted), you can either manually set suitable dimensions in Zim or add some CSS and/or Javascript to the HTML template to resize the images to automatically fit the viewport. Cool project btw!

**Reply by aloboa** [reply]
But Properties for these images in Zim state that the images must be reproduced 512 px wide, should this not
be thus enforced in the HTML? Not sure I can qualify this as a feature request or a bug (export should follow specifications in the zim document)

**Reply by introt** [reply]
Yes, the sizes should be enforced. I'm assuming you meant height as the two aren't the same width in either screenshot, but they do appear to maintain their dimensions - the second one just gets pushed onto the next row due to the index sidebar being wider in the web page version.

**Reply by introt** [reply]
No, sorry, the second one is indeed not as tall as the first one in the web version. Care to check if the html source has width or height attributes set? They should be in according to `dump_img` in zim/formats/html.py

**Reply by aloboa** [reply]
In Zim, I have the height set to 512 in both cases. 
The HTML states
 height="512" in the "<img...>"  field.
I think it could be a matter of the  Template (ZeroFiveEight, too wide a TOC), but most users
(as me) do not know how to modify templates. Perhaps more choices of well-designed templates are needed.
